app/ui/lib/Combobox.tsx

```diff
@@ -136,6 +143,17 @@ export const Combobox = ({
   }
   const zIndex = usePopoverZIndex()
   const id = useId()
+  const handleKeyDown = (e: React.KeyboardEvent<HTMLInputElement>, open: boolean) => {
+    // Prevent form submission when the user presses Enter inside a combobox.
+    // The combobox component already handles Enter keypresses to select items,
+    // so we only preventDefault when the combobox is closed.
+    if (e.key === 'Enter' && !open) {
+      e.preventDefault()
+      onEnter?.(e)
+    }
+    onKeyDown?.(e)
+  }
+
```

Comment:
Beautiful. If you inline this, you don't have to pass `open` because it's closed over. I think we could even do without the `onKeyDown` prop on `Combobox` if we're not using it.